samples/nrf9160/lwm2m_client/scripts/device.py

```diff
@@ -9,12 +9,59 @@
 import sys
 import argparse
 import logging
+import os
+import subprocess
+import serial
+import time
+import re
+
 from atclient import ATclient
+from serial.tools.list_ports import comports
+
+NRF91_VID = 4966
 
 class Device:
     """Generate requests for device using AT commands"""
-    def __init__(self, serial_path):
-        self.at_client = ATclient(serial_path)
+    def __init__(self):
+        self.sid = self.select_device()
+        self.com_port = self.get_com_ports(self.sid)[0]
+        self.sample_path = os.path.join("/".join(os.path.abspath(__file__).split("/")[:-3]), "at_client")
+
+        self.build()
+        self.flash()
+
+        self.wait_for_uart_string("The AT host sample started")
```

Comment:
This is otherwise looking good, but automatically building and flashing will be a breaking change.

Can you add those 3 lines under a command line argument.
This is because I have just tested that you can run this provisioning script against Asset Tracker, or any other app that accept AT commands. It does not have to be just the AT Host sample.